### ISSUE TITLE
Jetpack Dashboard shows Plugins menu as selected

### ISSUE THREAD

**Issue by cpapazoglou** [Automattic/wp-calypso#50713]
### Steps to reproduce the behavior

1. On Atomic, go to `/wp-admin/admin.php?page=jetpack#/dashboard`

#### What I expected to happen
Jetpack menu item to be selected

#### What actually happened
Plugins menu item was selected

#### Does this happen on simple or atomic sites or both?
Only on Atomic

![](https://cln.sh/dCEQ9X+)

**Comment by getdave** [comment]
Confirmed. Can replicate.

**Comment by getdave** [comment]
It looks like the menu slug is incorrect on WPAdmin. If in `projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php` I use `jetpack` the menu works as expected. What I'm not sure about is why we are doing the other thing...

```diff
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
index c62441fd1..42fd25da0 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
@@ -652,7 +652,8 @@ class Admin_Menu {
 
 		// TODO: Replace with proper SVG data url.
 		$jetpack_icon = "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 32 32' %3E%3Cpath fill='%23a0a5aa' d='M16,0C7.2,0,0,7.2,0,16s7.2,16,16,16s16-7.2,16-16S24.8,0,16,0z'%3E%3C/path%3E%3Cpolygon fill='%23fff' points='15,19 7,19 15,3 '%3E%3C/polygon%3E%3Cpolygon fill='%23fff' points='17,29 17,13 25,13 '%3E%3C/polygon%3E%3C/svg%3E";
-		$jetpack_slug = 'https://wordpress.com/activity-log/' . $this->domain;
+		$jetpack_slug = 'jetpack';
 
 		$this->add_admin_menu_separator( $position++, 'manage_options' );
 		add_menu_page( esc_attr__( 'Jetpack', 'jetpack' ), __( 'Jetpack', 'jetpack' ), 'manage_options', $jetpack_slug, null, $jetpack_icon, $position );

```

**Comment by getdave** [comment]
It looks to be that the parent file we've used doesn't marry up with the `$jetpack_slug`. I don't fully understand why we would have used `https://wordpress.com/activity-log/` for the Jetpack slug.

We can no longer patch this directly in Jetpack. Therefore I've created 628-gh-Automattic/wpcomsh which "fixes" things temporarily by correctly assigning the parent file to match the slug.

I also explored fixing the slug itself directly in Jetpack but we won't be able to deploy this to Atomic in time for launch

https://github.com/Automattic/jetpack/pull/19017

**Comment by mmtr** [comment]
> It looks like the menu slug is incorrect on WPAdmin. If in `projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php` I use `jetpack` the menu works as expected. What I'm not sure about is why we are doing the other thing...
> 
> ```diff
> diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
> index c62441fd1..42fd25da0 100644
> --- a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
> +++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
> @@ -652,7 +652,8 @@ class Admin_Menu {
>  
>  		// TODO: Replace with proper SVG data url.
>  		$jetpack_icon = "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 32 32' %3E%3Cpath fill='%23a0a5aa' d='M16,0C7.2,0,0,7.2,0,16s7.2,16,16,16s16-7.2,16-16S24.8,0,16,0z'%3E%3C/path%3E%3Cpolygon fill='%23fff' points='15,19 7,19 15,3 '%3E%3C/polygon%3E%3Cpolygon fill='%23fff' points='17,29 17,13 25,13 '%3E%3C/polygon%3E%3C/svg%3E";
> -		$jetpack_slug = 'https://wordpress.com/activity-log/' . $this->domain;
> +		$jetpack_slug = 'jetpack';
>  
>  		$this->add_admin_menu_separator( $position++, 'manage_options' );
>  		add_menu_page( esc_attr__( 'Jetpack', 'jetpack' ), __( 'Jetpack', 'jetpack' ), 'manage_options', $jetpack_slug, null, $jetpack_icon, $position );
> ```

This is really weird. We're using the `/activity-log` slug because that's usually the URL of the first submenu of the Jetpack menu, but it shouldn't really matter what we set there because we're migrating all the submenus using the `jetpack` slug to the new slug in https://github.com/Automattic/jetpack/blob/294ad77419b63293fde382ed6c4f268677dec72c/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L694 and https://github.com/Automattic/jetpack/blob/294ad77419b63293fde382ed6c4f268677dec72c/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L701-L706

I wonder if the problem is caused by the fact that the `jetpack#/dashboard` page is registered via the `jetpack_admin_menu` action:

https://github.com/Automattic/jetpack/blob/294ad77419b63293fde382ed6c4f268677dec72c/projects/plugins/jetpack/class.jetpack-admin.php#L42
https://github.com/Automattic/jetpack/blob/294ad77419b63293fde382ed6c4f268677dec72c/projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php#L56

Maybe we're changing the admin menu before the `jetpack_admin_menu` hook is triggered?

**Comment by mmtr** [comment]
I confirmed the `jetpack_admin_menu` is unrelated (it gets triggered within `admin_menu` before `reregister_menus` runs). Not sure why, but apparently the issue is that the `jetpack#/dashboard` page ends up with `parent_file` being the Plugins menu slug, rather than the Jetpack menu slug. This is avoided if we don't change the Jetpack menu slug to `/activity-log` but no idea why. This is certainly beyond my knowledge so let's ship that quick fix even if we don't know how that solves the issue.

**Comment by mmtr** [comment]
Found it! We have a bug in `add_plugins_menu` which uses an incorrect slug: https://github.com/Automattic/jetpack/blob/4285c543fac82809bd640485c071138076c7900e/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L515

It should check against `plugins.php`:

```diff
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
index 0231813e7..891c13b4e 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php
@@ -512,7 +512,7 @@ class Admin_Menu {
                add_filter(
                        'parent_file',
                        function ( $parent_file ) use ( $menu_slug ) {
-                               return 'jetpack' === $parent_file ? $menu_slug : $parent_file;
+                               return 'plugins.php' === $parent_file ? $menu_slug : $parent_file;
                        }
                );
        }
```

**Comment by getdave** [comment]
Deployed 628-gh-Automattic/wpcomsh to hotfix this.

**Comment by mmtr** [comment]
Ii will be also fixed in Jetpack 9.6 by https://github.com/Automattic/jetpack/pull/19017.
